src/Microsoft.ML.FastTree/FastTreeRanking.cs

```diff
@@ -343,6 +343,9 @@ protected override void CustomizedTrainingIteration(RegressionTree tree)
         /// <returns>standard test for the dataset</returns>
         private Test CreateStandardTest(Dataset dataset)
         {
+            if (dataset.MaxDcg.Length == 0)
```

Comment:
>dataset.MaxDcg.Length [](start = 16, length = 21)

Please replace `dataset.MaxDcg.Length` with `Utils.Size(dataset.MaxDcg)`. Under certain codepaths this array *may* be null. :P